libraries/freertos_plus/standard/freertos_plus_posix/test/iot_test_posix_utils.c

```diff
@@ -84,6 +97,31 @@ TEST( Full_POSIX_UTILS, UTILS_TimespecAdd )
     TEST_ASSERT_EQUAL_INT( -1, UTILS_TimespecAdd( NULL, &y, &xResult ) );
     TEST_ASSERT_EQUAL_INT( -1, UTILS_TimespecAdd( &x, NULL, &xResult ) );
 
+    /* Test with invalid parameter check when tv_sec < 0. */
+    if( isTimespecTvSecSigned == 1u )
+    {
+        x.tv_sec = -1;
+        TEST_ASSERT_EQUAL_INT( -1, UTILS_TimespecAdd( &x, &y, NULL ) );
+
+        x.tv_sec = 0;
+        y.tv_sec = -2;
+        TEST_ASSERT_EQUAL_INT( -1, UTILS_TimespecAdd( &x, &y, NULL ) );
+    }
+
+    /* Test with invalid parameter check when tv_nsec < 0. */
+    if( isTimespecTvSecSigned == 1u )
```

Comment:
Can be in the same if check as above.

Reply:
Updated this check to instead use the `isTimespecTvNsecSigned` global variable to pass negative values only when the `tv_nsec` member type is signed